tracing/tracing.go

```diff
@@ -37,14 +37,25 @@ func InstrumentHTTP(h http.Handler) http.Handler {
 		}
 		ctx = slog.NewContext(ctx, log)
 
-		log.Debug("handling request", "url", req.URL, "method", req.Method)
+		httpReq := map[string]any{
```

Comment:
struct should also work no ? maybe that would be more clear/safe since it will be clear which fields are which type etc (there is no dynamic nature here AFAIK, only optional fields that can be ommited etc).

Reply:
The problem is, if it's a struct defined inside of tracing package, I'll not be able to read it on the slog package.

Reply:
it doesn't convert the fields defined in the struct to a map internally ? I kinda expected that..but never tested since this is quite advanced.... hmm maybe it keeps the original struct and the JSON handler itself transforms it...so yeah it would not be available as a map...and then reflection is used on the JSON handler...yeah makes sense, that is why the TextHandler only prints the struct itself (probably using String()/same mechanism of the fmt.Print) and the JSON handler does something else.

All good then, disregard =P

Reply:
Although there seems to be an implicit protocol between the tracing pkg and the slog pkg..not sure keeping this implicit/dynamic is the best way to handle this...but maybe it is...hmm

Reply:
Also a point!

Reply:
yeah..changes here will break there..silently....and adding new things will always be coordinated anyway (or it won't work properly for gcloud formatting)...hmm, we can assess this further later...really not sure about this